spacelift/data_tool_versions.go

```diff
@@ -0,0 +1,99 @@
+package spacelift
+
+import (
+	"context"
+	"slices"
+
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+)
+
+func dataToolVersions() *schema.Resource {
+	return &schema.Resource{
+		Description: "Lists supported versions for a given tool.",
+		ReadContext: dataToolVersionsRead,
+		Schema: map[string]*schema.Schema{
+			"tool": {
+				Type:             schema.TypeString,
+				Description:      "The tool to get a list of supported versions for. This can be one of `KUBECTL`, `OPEN_TOFU`, `TERRAFORM_FOSS`, or `TERRAGRUNT`.",
+				ValidateDiagFunc: dataToolVersionsValidateInput,
+				Required:         true,
+			},
+			"versions": {
+				Type:        schema.TypeList,
+				Description: "Supported versions of the given tool.",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataToolVersionsValidateInput(i interface{}, p cty.Path) diag.Diagnostics {
+	tool := i.(string)
+
+	validTools := []string{
+		"KUBECTL",
+		"OPEN_TOFU",
+		"TERRAFORM_FOSS",
+		"TERRAGRUNT",
+	}
+
+	if !slices.Contains(validTools, tool) {
+		return diag.Errorf("tool must be one of %v", validTools)
+	}
+
+	return nil
+}
+
+func dataToolVersionsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	d.SetId("spacelift-versions")
```

Comment:
Does this cause issues if you define multiple tool version data sources in the same terraform state? If so maybe use the tool as the ID?

Reply:
Damn - I should have read the PR comments first - I can see you tested that. Fair enough.

Reply:
Cool - yeah unfortunately from a quick search I wasn't sure whether this would matter or not, but I agree it's probably safer to make it unique.